tests/unit/Email.spec.js

```diff
@@ -0,0 +1,74 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import Email from '@/components/Email.vue'
+import VueRouter from 'vue-router'
+
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+const router = new VueRouter()
+
+describe('Email', () => {
+  it('should render Email', () => {
+    const email = shallowMount(Email, {localVue, router})
+    expect(email.exists()).toBe(true)
+    expect(email.find('h1').text()).toBe('Enter your email to get started!')
+    expect(email.find('input').attributes('placeholder')).toBe('example@gmail.com')
+    expect(email.find('button').text()).toBe('SUBMIT')
+    expect(email.find('h3').exists()).toBe(true)
+  })
+
+  it('should accept user input for email', () => {
+    const email = shallowMount(Email, {localVue, router})
+    const input = email.find('input')
+    expect(input.element.value).toBe('')
+    input.setValue('boneyard@yahoo.com')
+    expect(input.element.value).toBe('boneyard@yahoo.com')
+  })
+
+  it('should have a disabled button if input is empty', () => {
+    const email = shallowMount(Email, {localVue, router,
+      data() {
+        return {
+          message: ''
+        }
+      }})
+    const button = email.find('button')
+    expect(button.attributes('class')).toBe('disabled')
+  })
+
+  it.skip('should show error message', async () => {
+    const email = shallowMount(Email, {localVue, router,
+      computed: {
+        isInvalid() {
+          return true
+        }
+      },
+      data() {
+        return {
+          email: 'boneyard@yahoo.com'
+        }
+      }
+      })
+      const spy = jest.spyOn(email.vm, 'submitEmail')
+      const button = email.find('button') 
+      expect(email.find('input').element.value).toBe('boneyard@yahoo.com')
+      expect(button.attributes('class')).toBe('')
```

Comment:
I might be grasping at straws here, but if you're testing the method for sad path here, wouldn't you want to mock the email data with the wrong format? Like `'boneyard@yahoocom'`? I see the `isInvalid` value is true and I would think the method should fire regardless, but I wonder if the email is throwing the test off? I know we have some other tests in our suite that have the same issue with `setMethod()` being used but deprecated so I will try and do some research on this as well.

Reply:
Good point, Stacy.  I did experiment with the `data` and put in an invalid format at some point when I was trying things out.  I think I found it wasn't relevant to the firing of `submitEmail( )`, but for this specific test, it would make sense to put in an incorrect email so that the proper error message could show up.